### PR TITLE
[fix](Nereids)plan translator lost other conjuncts on hash join node

### PR DESCRIPTION
**Pull request by morrySnow**
# Proposed changes

## Problem summary

In the earlier PR #11812 , we split join condition into two parts: hash join conjuncts and other condition. But we forgot to translate other condition into other conjuncts in HashJoinNode of legacy planner. So we get wrong result if query has other condition on join node. Such as:

SELECT * FROM lineorder INNER JOIN part ON lo_partkey = p_partkey WHERE lo_orderkey > p_size;

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
LGTM

**Review by 924060929 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -378,8 +379,16 @@ public PlanFragment visitPhysicalHashJoin(PhysicalHashJoin<Plan, Plan> hashJoin,
                 .map(e -> ExpressionTranslator.translate(e, context))
                 .collect(Collectors.toList());
 
+        List<Expr> otherJoinConjuncts = hashJoin.getOtherJoinCondition()
+                .map(ExpressionUtils::extractConjunction)
+                .orElseGet(Lists::newArrayList)
+                .stream()
+                .filter(e -> !(e.equals(BooleanLiteral.TRUE)))
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:386]: do we have to consider const conjuncts here?
`TRUE`  is a special case of const conjunct. 
There are some other conjucts, like `1=1` or `FALSE`.
I think there should be a rule to clean them before translate.
So I think we do not need to filter them here.

Comment (jackwener) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:386]: > do we have to consider const conjuncts here? `TRUE` is a special case of const conjunct. There are some other conjucts, like `1=1` or `FALSE`. I think there should be a rule to clean them before translate. So I think we do not need to filter them here.

It's should be consider by other rule

**Review by jackwener — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java (hunk 1 above)
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:386]: So, filter(...) is useless. It is better to remove this call

Comment (jackwener) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:386]: Tmp way, add TODO is enough.

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:386]: @jackwener say all what i want say

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:388]: add TODO

**Review by 924060929 — APPROVED** [review]

**Pull request merged**